samples/excel/89-preview-apis/range-copyfrom.yaml

```diff
@@ -0,0 +1,194 @@
+                sheet.getRange("G1").copyFrom("A1:E1");
+                await context.sync();
+            });
+        }
+
+        async function copyFormula() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.getRange("F2").values = [["Copied Formula"]];
+                sheet.getRange("F2").format.autofitColumns();
+
+                // copy a range preserving the formulas
+                // note: non-formula values are copied over as is
+                sheet.getRange("G2").copyFrom("A1:E1", Excel.RangeCopyType.formulas);
+                await context.sync();
+            });
+        }
+
+        async function copyFormulaResult() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.getRange("F3").values = [["Copied Formula Result"]];
+                sheet.getRange("F3").format.autofitColumns();
+
+                // copy the resulting value of a formula
+                sheet.getRange("G3").copyFrom("E1", Excel.RangeCopyType.values);
+                await context.sync();
+            });
+        }
+
+        async function copySingleAcrossRange() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.getRange("F4").values = [["Single Source"]];
+                sheet.getRange("F4").format.autofitColumns();
+
+                // copy a single cell across an entire range
+                sheet.getRange("G4:K4").copyFrom("A1", Excel.RangeCopyType.values);
+                await context.sync();
+            });
+        }
+
+        async function copyOnlyFormat() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.getRange("F5").values = [["Copied Formatting"]];
+                sheet.getRange("F5").format.autofitColumns();
+
+                // copy only the formatting of the cells
+                sheet.getRange("G5").copyFrom("A1:E1", Excel.RangeCopyType.formats);
+                await context.sync();
+            });
+        }
+
+        async function skipBlanks() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+                sheet.getRange("F6").values = [["Copied Without Blanks"]];
+                sheet.getRange("F6").format.autofitColumns();
+
+                // fill the destination range so we can see the blank being skipped
+                sheet.getRange("G6:K6").values = [["Old Data", "Old Data", "Old Data", "Old Data", "Old Data"]]
+
+                // copy a range, omitting the blank cells so existing data is not overwritten in those cells
+                sheet.getRange("G6").copyFrom("A1:E1", Excel.RangeCopyType.all, true, false);
```

Comment:
Since Boolean parameters aren't self-documenting its a good idea to put the parameter name in an inline comment. So this line would read:
`eet.getRange("G6").copyFrom("A1:E1", Excel.RangeCopyType.all, true /* skipBlanks */, false /* transpose */); `